code/modules/hydroponics/trays/tray.dm

```diff
@@ -151,17 +151,23 @@
 	return ..()
 
 /obj/machinery/portable_atmospherics/hydroponics/attack_ghost(var/mob/abstract/observer/user)
+	..()
+	if(!harvest)
+		to_chat(user, SPAN_WARNING("\The [src]'s plant isn't ready to harvest yet."))
+		return
 
 	if(!(harvest && seed && seed.has_mob_product))
 		return
```

Comment:
This should be changed so the check for `seed` and `seed.has_mob_product` occurs first with a silent exit.

Then there should be the check for `harvest` with a message to the user in case of failure.